### ISSUE TITLE
Unable to render leaflet map created with mapBliss

### ISSUE THREAD

**Issue by benyamindsmith** [posit-dev/positron#3756]
## Positron Version:

Positron Version: 2024.06.1 (system setup) build 2024.06.1-27
Code - OSS Version: 1.90.0
Commit: a893e5b282612ccb2200102957ac38d3c14e5196
Date: 2024-06-26T01:33:58.809Z
Electron: 29.4.0
Chromium: 122.0.6261.156
Node.js: 20.9.0
V8: 12.2.281.27-electron.0
OS: Windows_NT x64 10.0.22631

## Steps to reproduce the issue:

1.  Install my mapBliss package `devtools::install_github("benyamindsmith/mapBliss")` (repository here: https://github.com/benyamindsmith/mapBliss)
2. Run the following code: 

```r
mp <- mapBliss::plot_hybrid_route_flex(
  addresses = c("Toronto, ON, Canada",
                "Buffalo, NY, USA"),
  how = c("car")
)

mp
```

![image](https://github.com/posit-dev/positron/assets/46410142/d9b0c106-f905-4537-a1fb-d3e981fd8959)

## What did you expect to happen?

In RStudio I get the following output in the Viewer Pane: 

![image](https://github.com/posit-dev/positron/assets/46410142/083446c2-a216-4aac-ae4f-cbd078d60ea5)

## Were there any error messages in the output or Developer Tools console?

![image](https://github.com/posit-dev/positron/assets/46410142/7d727e03-d37c-409c-8e90-3eb22cd45918)

**Comment by juliasilge** [comment]
Using leaflet itself does work, such as with this:

```r
library(leaflet)

leaflet() |> 
    addTiles() |> 
    addMarkers(lng = -111.90, lat = 40.76)
```

But I can confirm problems with the example mapBliss code. I see this in the Plots pane (no background tiles):

![Screenshot 2024-06-30 at 3 41 53 PM](https://github.com/posit-dev/positron/assets/12505835/094d2bbf-7274-438b-8899-a1fd27a4a07f)

Related to #2023

**Comment by benyamindsmith** [comment]
@juliasilge it looks like for the mapBliss tiles to render in positron, they need to be specified as `https://` as opposed to `//{s}.`

```r
mp <- mapBliss::plot_hybrid_route_flex(
  addresses = c("Toronto, ON, Canada",
                "Buffalo, NY, USA"),
  how = c("car"),
  mapBoxTemplate = "https://tile.openstreetmap.org/{z}/{x}/{y}.png"
)

mp
```

![image](https://github.com/posit-dev/positron/assets/46410142/db99cb3c-b0a7-4789-9e80-a81ee1d0d260)

The `mapBoxTemplate` argument simply provides tiles to the `addTiles` arg in leaflet (line of code [here](https://github.com/benyamindsmith/mapBliss/blob/605f6ca1b8d0122c16cdc934f011208ae4e01a9a/R/plot_hybrid_route_flex.R#L106C5-L106C47)). So it seems strange that the behavior is different.

**Comment by benyamindsmith** [comment]
To refine the issue, the issue would be with the `addTiles()` argument with leaflet.  

i.e.

```r
leaflet() |> 
    addTiles(urlTemplate = "//{s}.tile.openstreetmap.org/{z}/{x}/{y}.png" ) |> 
    addMarkers(lng = -111.90, lat = 40.76)
```

![image](https://github.com/posit-dev/positron/assets/46410142/658dd7dd-ccd0-43bb-b280-b831180a3210)


Not sure what the issue could be but it is indeed strange.

**Comment by ElianHugh** [comment]
is it possible that this is an issue with CSP headers?

**Comment by benyamindsmith** [comment]
@ElianHugh  that's a good guess. It also lines up with the issues shown in the Developer Tools Console. I wonder if there are other issues with CSPs beyond leaflet rendering.

**Comment by jmcphers** [comment]
As noted in #3949, the problem also reproduces with `addProviderTiles` (another way of loading tiles from an external source; as in https://github.com/posit-dev/positron/issues/3756#issuecomment-2198816489).

**Comment by jmcphers** [comment]
Should be addressed in #4151:

<img width="1153" alt="image" src="https://github.com/user-attachments/assets/e6031ad6-f2b8-406c-be25-bc83bdb666ae">

**Comment by testlabauto** [comment]
### Verified Fixed
    Positron Version(s) : 2024.07.0-107
    OS Version          : OSX

### Test scenario(s) 

mp <- mapBliss::plot_hybrid_route_flex(
  addresses = c("Toronto, ON, Canada",
                "Buffalo, NY, USA"),
  how = c("car")
)

mp

Link(s) to TestRail test cases run or created: 
N/A
